torch/csrc/jit/passes/onnx.cpp

```diff
@@ -287,8 +299,13 @@ void NodeToONNX(
           ONNXShapeTypeInference(const_node, empty_params_dict, opset_version);
           env[old] = const_node->output();
         } else {
-          outputs[i]->setType(
-              MergeInferredType(old->type(), outputs[i]->type()));
+          auto input_reliable = IsShapeReliable(outputs[i]->node());
+          auto mergeTypePair =
```

Comment:
would it make sense to create a new ONNXType inheriting from jit Type, such that the reliable and etc information can be stored naturally?

Reply:
nit: can have
```c++
TypePtr mergedType;
bool inferred;
std::tie(mergedType, inferred) = MergeInferredType(...);
```

hopefully pytorch moves to c++17 soon then we can have `auto [mergedType, inferred] = MergedInferredType(...);` ...

edit: this applies to `MergeInferredTypeAndSetMap` too

Reply:
why not use `MergeInferredTypeAndSetMap` here as well?

Reply:
done with the `std::tie` solution.